corgi/collectors/yum.py

```diff
@@ -0,0 +1,149 @@
+import logging
+from collections import defaultdict
+
+from corgi.collectors.brew import Brew
+from corgi.collectors.pulp import Pulp
+from corgi.tasks.common import run_external
+
+logger = logging.getLogger(__name__)
+
+
+class Yum:
+    """Get NVR / NEVRA / NSVCA data for an arbitrary repo using DNF commands.
+    Usually the Pulp collector should be used instead.
+    This Yum collector is only needed for community repos,
+    or Red Hat repos which aren't tracked in Pulp (like the ones for RHIVOS).
+    """
+
+    def __init__(self):
+        self.brew = Brew()
+
+    @staticmethod
+    def filter_by_repos(command: list[str], repos: list[str]) -> list[str]:
+        """Take some DNF command and run it against only the repos specified."""
+        repo_id = 1
+        for repo_name in repos:
+            command.append(f"--repofrompath={repo_id},{repo_name}")
+            command.append(f"--repoid={repo_id}")
+            repo_id += 1
+        return command
+
+    def get_modules(
+        self, module_names: list[str], repos: list[str]
+    ) -> list[dict[str, dict[str, list[str]]]]:
+        """Helper function to get specific data for modules in particular Yum repos."""
+        command = ["dnf", "--quiet", "module", "info"]
+        command = self.filter_by_repos(command, repos)
+        for module_name in module_names:
+            command.append(module_name)
+        result, output = run_external(command)
+
+        modules_lines = []
+        start_line_no = 0
+        for line_no in range(len(output)):
+            if output[line_no] == "":
+                modules_lines.append(output[start_line_no:line_no])
+                start_line_no = line_no + 1
+
+        return [self.parse_module(lines) for lines in modules_lines]
+
+    def find_modules_from_yum_repos(
+        self, repos: list[str]
+    ) -> list[dict[str, dict[str, list[str]]]]:
+        """List and collect info for all RPM modules in particular Yum repos."""
+        command = ["dnf", "--quiet", "module", "list"]
```

Comment:
I probably should, thanks for the link! It's my first time working with repoquery / dnf in general, so Jason suggested copying the Deptopia logic. And obviously that's written in go so doesn't directly interface with the DNF Python library. I'll try to get to this after finishing tests.

Reply:
Don't feel like you have to spend tons of time on it though. If the current way works then it's fine as is. Only if it's trivial to use DNF through its Python API it might be worth it to refactor this.

Reply:
Quick look doesn't provide any obvious way to list modules in a repo, or limit a query to a particular repo (I see .available() but that lists packages in all repos). So going to leave as-is for now. If I missed the obvious and there is some way to do this, please feel free to file a backlog ticket.